src/kernels/ipywidgets/commonMessageCoordinator.ts

```diff
@@ -106,10 +115,45 @@ export class CommonMessageCoordinator {
                 (m) => {
                     traceInfoIfCI(`${ConsoleForegroundColors.Green}Widget Coordinator received ${m.type}`);
                     this.onMessage(m.type, m.payload);
-
+                    const kernel = this.kernelProvider.get(this.document.uri);
                     // Special case the WidgetManager loaded message. It means we're ready
                     // to use a kernel. (IPyWidget Dispatcher uses this too)
                     if (m.type === IPyWidgetMessages.IPyWidgets_Ready) {
+                        if (kernel?.kernelConnectionMetadata.kind === 'startUsingRemoteKernelSpec') {
```

Comment:
Should `startUsingRemoteKernelSpec` be elsewhere? Like in a `constants` kind of file?

Reply:
No it's an enum. The compiler will check if you type it in wrong. Here's an example of me typing it in wrong:

![image](https://user-images.githubusercontent.com/19672699/176969566-576c96b9-fd65-4197-b51a-3b39ea770106.png)